### ISSUE TITLE
Add `until(cond; next)` builtin

### ISSUE THREAD

**Issue by pkoppstein** [stedolan/jq#639]
The following "until" control structure fills a gap in the current (1 Dec 2014) offering of builtins, most
obviously because it emits at most one result, whereas "recurse" and "while" emit arbitrarily many values:

```
# As soon as "condition" is true, then emit . and stop:
def do_until(condition; next):
  def u: if condition then . else (next|u) end;
  u;
```

Thus, do_until avoids the inelegance and inefficiency of having to use last(g) (available in jq>1.4) in order to obtain the last result.  

Another salient point is that sometimes one simply wants to loop until a value is found that satisfies a condition. This is not so easily accomplished using the existing while(cond;_), simply because all the values it emits satisfy cond.

That is, do_until(cond; next) is NOT equivalent to last(while(cond|not; next)).

In summary, there are at least three reasons for adding do_until (however named) to builtin.c:

1) expressiveness
2) efficiency
3) elegance

The practical utility of do_until is illustrated by various examples such as at
http://rosettacode.org/wiki/Sorting_algorithms/Comb_sort#jq 

In this posting, I've used the name "do_until" to emphasize the difference between it and the other "loop" control structures, but a case can obviously be made for calling it simply "until".
